internal/parser/scanner/states.go

```diff
@@ -356,231 +406,45 @@ var keywordsWithW map[string]token.Type = map[string]token.Type{
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithQ[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanRKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithR[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithR[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanSKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithS[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithS[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanTKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithT[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithT[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-
-func scanUKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithU[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithU[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
-}
-func scanVKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithV[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithV[input], input, s)
-	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
+func (s *scanner) scanLiteral() token.Token {
+	return s.unexpectedRune("nil")
 }
 
-func scanWKeyword(s *scanner) token.Token {
-	input := string(s.input[s.start:s.seekNext(s.start)])
-	if _, ok := keywordsWithW[input]; ok {
-		s.acceptString(input)
-		return createToken(s.line, s.col, s.start, s.pos, keywordsWithW[input], input, s)
+func (s *scanner) scanSpace() token.Token {
+	if s.input[s.start] == '\n' {
+		s.line++
+		s.col = 1
 	}
-	// return an error that this doesnt exist
-	return createToken(s.line, s.col, s.start, s.pos, token.KeywordSelect, string(s.input[s.start:s.pos]), s)
+	s.pos++
+	s.start = s.pos
+	return nil
```

Comment:
Don't return nil. Either return a token or make the function return nothing (no return type).